src/framework/components/layout-group/component.js

```diff
@@ -0,0 +1,214 @@
+        // to self or to children - covers cases where they are not already
+        // present at the point when this component is constructed.
+        system.app.systems.element.on('add', this._onElementComponentAdd, this);
+        system.app.systems.element.on('beforeremove', this._onElementComponentRemove, this);
+        system.app.systems.layoutchild.on('add', this._onLayoutChildComponentAdd, this);
+        system.app.systems.layoutchild.on('beforeremove', this._onLayoutChildComponentRemove, this);
+
+        this._scheduleReflow();
+    };
+    LayoutGroupComponent = pc.inherits(LayoutGroupComponent, pc.Component);
+
+    pc.extend(LayoutGroupComponent.prototype, {
+        _isSelfOrChild: function(entity) {
+            return (entity === this.entity) || (this.entity.children.indexOf(entity) !== -1);
+        },
+
+        _listenForResizeEvents: function(target, onOff) {
+            if (target.element) {
+                target.element[onOff]('resize', this._onResize, this);
+                target.element[onOff]('set:pivot', this._onSetPivot, this);
+            }
+
+            if (target.layoutchild) {
+                target.layoutchild[onOff]('resize', this._onResize, this);
+            }
+        },
+
+        _onElementComponentAdd: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.on('resize', this._onResize, this);
+            }
+        },
+
+        _onElementComponentRemove: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.off('resize', this._onResize, this);
+            }
+        },
+
+        _onLayoutChildComponentAdd: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.on('resize', this._onResize, this);
+            }
+        },
+
+        _onLayoutChildComponentRemove: function(entity, component) {
+            if (this._isSelfOrChild(entity)) {
+                component.off('resize', this._onResize, this);
+            }
+        },
+
+        _onResize: function() {
+            this._scheduleReflow();
+        },
+
+        _onSetPivot: function() {
+            this._scheduleReflow();
+        },
+
+        _onChildInsert: function(child) {
+            this._listenForResizeEvents(child, 'on');
+            this._scheduleReflow();
+        },
+
+        _onChildRemove: function(child) {
+            this._listenForResizeEvents(child, 'off');
+            this._scheduleReflow();
+        },
+
+        _scheduleReflow: function() {
+            if (!this._isPerformingReflow) {
+                this.system.scheduleReflow(this);
+            }
+        },
+
+        reflow: function() {
+            var container = getElement(this.entity);
+            var elements = this.entity.children.filter(getElement).map(getElement);
+
+            if (!container || elements.length === 0) {
```

Comment:
@daredevildave should we also check whether the `LayoutGroup` and its `Entity` are enabled here before proceeding with the reflow?

Reply:
Agreed